velox/docs/functions/spark/string.rst

```diff
@@ -427,3 +427,25 @@ String Functions
     Returns string with all characters changed to uppercase. ::
 
         SELECT upper('SparkSql'); -- SPARKSQL
+
+.. spark:function:: varchar_type_write_side_check(string, limit) -> varchar
+
+    Removes trailing space characters (ASCII 32) that exceed the maximum Unicode string length ``limit`` in characters.
+    This function will trim at most the length of ``string`` in characters - ``limit`` space characters from the end of this string.
+    Throws exception when the string still exceeds ``limit`` after trimming trailing spaces or when ``limit`` is negative.
+    Note: This function is not directly callable in SQL, but internally used by SparkSQL for length check when writing to
+    varchar type columns. ``limit=0`` is technically valid for this function but SparkSQL does not support ``VARCHAR(0)``
+    in DDL. This case exists primarily for internal consistency. ::
+
+        -- Example SparkSQL usage that triggers the function (function is not called directly in SQL).
+        create table src(id string) stored as parquet;
+        create table tgt(id varchar(3)) stored as parquet;
+        insert into src values ('abc '); -- ascii string with characters length of 4.
```

Comment:
Examples like `varchar_type_write_side_check(xxx, xxx) -- xxx` would be good to illustrate the semantics, seeing https://github.com/facebookincubator/velox/pull/13034#discussion_r2048600431.

Reply:
I assume current example could hardly help the readers understand the function semantics, and it would be good for eamples like `varchar_type_write_side_check(xxx, xxx) -- xxx` as you have clarified this is a Spark internal function. Thanks.